packages/tx/src/eip2930Transaction.ts

```diff
@@ -0,0 +1,385 @@
+          value: new BN(value),
+          data: data ?? emptyBuffer,
+          accessList: accessList ?? emptyBuffer,
+          v: v !== undefined ? new BN(v) : undefined, // EIP2930 supports v's with value 0 (empty Buffer)
+          r: r !== undefined && !r.equals(emptyBuffer) ? new BN(r) : undefined,
+          s: s !== undefined && !s.equals(emptyBuffer) ? new BN(s) : undefined,
+        },
+        opts ?? {}
+      )
+    } else {
+      throw new Error(
+        'Invalid EIP-2930 transaction. Only expecting 8 values (for unsigned tx) or 11 values (for signed tx).'
+      )
+    }
+  }
+
+  public constructor(txData: TxData, opts: TxOptions = {}) {
+    const { chainId, nonce, gasPrice, gasLimit, to, value, data, accessList, v, r, s } = txData
+
+    super({ nonce, gasPrice, gasLimit, to, value, data }, opts)
+
+    // EIP-2718 check is done in Common
+    if (!this.common.isActivatedEIP(2930)) {
+      throw new Error('EIP-2930 not enabled on Common')
+    }
+
+    // check  the type of AccessList. If it's a JSON-type, we have to convert it to a buffer.
+
+    let usedAccessList
+    if (accessList && isAccessList(accessList)) {
+      this.AccessListJSON = accessList
+
+      const newAccessList: AccessListBuffer = []
+
+      for (let i = 0; i < accessList.length; i++) {
+        const item: AccessListItem = accessList[i]
+        //const addItem: AccessListBufferItem = []
+        const addressBuffer = toBuffer(item.address)
+        const storageItems: Buffer[] = []
+        for (let index = 0; index < item.storageKeys.length; index++) {
+          storageItems.push(toBuffer(item.storageKeys[index]))
+        }
+        newAccessList.push([addressBuffer, storageItems])
+      }
+      usedAccessList = newAccessList
+    } else {
+      usedAccessList = accessList ?? []
+      // build the JSON
+      const json: AccessList = []
+      for (let i = 0; i < usedAccessList.length; i++) {
+        const data = usedAccessList[i]
+        const address = bufferToHex(data[0])
+        const storageKeys: string[] = []
+        for (let item = 0; item < data[1].length; item++) {
+          storageKeys.push(bufferToHex(data[1][item]))
+        }
+        const jsonItem: AccessListItem = {
+          address,
+          storageKeys,
+        }
+        json.push(jsonItem)
+      }
+      this.AccessListJSON = json
+    }
+
+    this.chainId = chainId ? new BN(toBuffer(chainId)) : new BN(this.common.chainId())
+    this.accessList = usedAccessList
+    this.v = v ? new BN(toBuffer(v)) : undefined
+    this.r = r ? new BN(toBuffer(r)) : undefined
+    this.s = s ? new BN(toBuffer(s)) : undefined
+
+    if (!this.chainId.eq(new BN(this.common.chainId().toString()))) {
+      throw new Error('The chain ID does not match the chain ID of Common')
+    }
+
+    if (this.v && !this.v.eqn(0) && !this.v.eqn(1)) {
+      throw new Error('The y-parity of the transaction should either be 0 or 1')
+    }
```

Comment:
If we decide to move up this check should then also be in `BaseTransaction`.

Reply:
This specific check? `LegacyTransaction` can have any `v` value, it depends on the chain id? So if we would move this exact check to `BaseTransaction`, we break `LegacyTransaction`.